openslides_backend/presenter/search_users_by_name_email.py

```diff
@@ -0,0 +1,143 @@
+from typing import Any, Dict, List, Tuple
+
+import fastjsonschema
+
+from ..action.actions.user.user_scope_permission_check_mixin import UserScope
+from ..permissions.management_levels import (
+    CommitteeManagementLevel,
+    OrganizationManagementLevel,
+)
+from ..permissions.permission_helper import (
+    has_committee_management_level,
+    has_organization_management_level,
+    has_perm,
+)
+from ..permissions.permissions import Permissions
+from ..shared.exceptions import MissingPermission, PresenterException
+from ..shared.filters import FilterOperator, Or
+from ..shared.patterns import Collection, FullQualifiedId
+from ..shared.schema import schema_version
+from .base import BasePresenter
+from .presenter import register_presenter
+
+search_users_by_name_email_schema = fastjsonschema.compile(
+    {
+        "$schema": schema_version,
+        "type": "object",
+        "title": "search_users_by_name_email",
+        "description": "get lists of id, first-, last-name and email for tuples of exact (username, emails)-tuples.",
+        "properties": {
+            "permission_scope": {
+                "type": "array",
+                "prefixItems": [
+                    {
+                        "enum": [1, 2, 3]
+                    },  # Permission scopes Meeting, Committee, Organization
+                    {"type": "number"},  # Id of object, 1 for organization
+                ],
+            },
+            "search": {
+                "type": "array",
+                "minItems": 1,
+                "items": {
+                    "type": "array",
+                    "minItems": 2,
+                    "maxItems": 2,
+                    "prefixItems": [
```

Comment:
Why do you use this strange tuple notation? I would just use an array of objects like
```
[{
  "username": "xy",
  "email": "x@y",
}]
```
I had to look up what `prefixItems` even does :D

Reply:
On another note, we do not need to seperate between email and username. The use case is that the admin enters the email OR username into ONE input field, which is then sent to the server. The client does not have the possibility to differentiate between username and email and it is not needed.

Also, on the same note, an array is unnecessary since only one input will ever be sent at a time. But if you want to support multiple inputs I guess that's fine, too (just see the other comment below).